cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/TestNativeImageCommandTest.java

```diff
@@ -66,20 +71,30 @@ public void testNativeImageTestsWithAdditionalArgs() throws IOException {
         Path projectPath = this.testResources.resolve("validProjectWithTests");
         System.setProperty(ProjectConstants.USER_DIR, projectPath.toString());
         TestCommand testCommand = new TestCommand(projectPath, printStream, printStream, false, true, "-H:Name=foo");
-        new CommandLine(testCommand).parse();
-        testCommand.execute();
+        new CommandLine(testCommand).parseArgs();
+        try {
+            testCommand.execute();
+        } catch (BLauncherException e) {
+            readOutput(false);
+            Assert.fail(e.getDetailedMessages().get(0));
+        }
         String buildLog = readOutput(true);
-        Assert.assertTrue(buildLog.contains("Generating 'foo' (executable)"));
+        Assert.assertTrue(buildLog.contains("Generating 'foo' (executable)"), buildLog);
     }
 
     @Test(description = "Test function mocking")
     public void testNativeFunctionMockTests() throws IOException {
         Path projectPath = this.testResources.resolve("mockFunctionNative");
         System.setProperty(ProjectConstants.USER_DIR, projectPath.toString());
-        TestCommand testCommand = new TestCommand(projectPath, printStream, printStream, false, true, "");
+        TestCommand testCommand = new TestCommand(projectPath, System.out, System.out, false, true, "");
```

Comment:
@Shadow-Devil Locally I couldn't reproduce any compilation error for this test case (https://scans.gradle.com/s/3uvqcaqdxieqs/tests/task/:ballerina-cli:test/details/io.ballerina.cli.cmd.TestNativeImageCommandTest/testNativeFunctionMockTests?top-execution=1). But I got assertion error.  We pass a custom `printStream` and get the output printed to that. we use that for the assertions. Passing `System.out`  makes `buildLog` an empty string.  Assertions fail. Shall we revert it back to original logic?

Reply:
Oh, then thats my fault. I tried to use System.out to debug the compilation logs but forgot to switch it back. I will revert this change

Reply:
For some reason, this still fails on CI

Reply:
@Thevakumar-Luheerathan Now I got the exact error message that happens in CI for the testcase:
```
java.lang.AssertionError: io.ballerina.cli.launcher.BLauncherException	
[error: compilation contains errors]	
Compiling source	
	bar/winner:0.1.0	
ERROR [tests/tes.bal:(3:1,3:21)] cannot resolve module 'ballerina/os'	
ERROR [tests/tes.bal:(18:23,18:39)] undefined function 'getUserHome'	
ERROR [tests/tes.bal:(18:23,18:39)] undefined module 'os'	
ERROR [modules/util2/tests/tes.bal:(3:1,3:21)] cannot resolve module 'ballerina/os'	
ERROR [modules/util2/tests/tes.bal:(7:26,7:42)] undefined function 'getUserHome'	
ERROR [modules/util2/tests/tes.bal:(7:26,7:42)] undefined module 'os'	
ERROR [modules/util2/tests/tes.bal:(8:26,8:42)] undefined function 'getUserHome'	
ERROR [modules/util2/tests/tes.bal:(8:26,8:42)] undefined module 'os'	
ERROR [modules/util/tests/lib_test.bal:(5:1,5:21)] cannot resolve module 'ballerina/os'	
ERROR [modules/util/tests/lib_test.bal:(29:23,29:39)] undefined function 'getUserHome'	
ERROR [modules/util/tests/lib_test.bal:(29:23,29:39)] undefined module 'os'	
ERROR [modules/util/tests/lib_test2.bal:(3:1,3:21)] cannot resolve module 'ballerina/os'	
ERROR [modules/util/tests/lib_test2.bal:(11:23,11:39)] undefined function 'getUserHome'	
ERROR [modules/util/tests/lib_test2.bal:(11:23,11:39)] undefined module 'os'
```
Could you please help further with this strange issue? Thank you in advance!

Reply:
I would like to get this PR merged soon, so I would suggest ignoring this failing test case for now and create a new Issue for it. In my opinion, this is still better than the current behavior where the test is simply not executed because the runtime exits. See also this Build Scan of master where this testcase is not executed: https://scans.gradle.com/s/vth7wqxhkb6ue/tests/overview